django/urls/resolvers.py

```diff
@@ -338,11 +342,11 @@ def match(self, path):
         # If this is an endpoint, the path should be exactly the same as the
         # route.
         elif self._is_endpoint:
-            if self._route == path:
+            if str(route) == path:
                 return "", (), {}
-        # If this isn't an endpoint, the path should start with the route.
```

Comment:
This comment should not be removed, unless there is a reason to? A detailed explanation could help the reviewers understand the motivation for this removal.